Implement pre_login_flow & login_flow to simulate the real open/close application.

Tkd-Alex opened the pull request:
- Burped Android Instagram app and analyzed all request.
https://github.com/itsMoji/Instagram_SSL_Pinning
- Try to simulate all **login_flow** and open/close app, I hope It can be help  with action blocked. The **loginFlow** It's implements also here: 
https://github.com/mgp25/Instagram-API/blob/master/src/Instagram.php#L941
- Fix cookie file. Now the cookie It's a .json and also sore all uuid s generate randomly.
- Added other X-IG-fields in request header.
- **!!!** Update save/like/comment with other field founded during the burped analisys
- Other minor fix.

Thanks to @maxdevblock .

maxdevblock committed:
Fix 'id', 'phone_id' and 'device_id'

Tkd-Alex committed:
Merge pull request #1 from maxdevblock/patch-1

Fix 'id', 'phone_id' and 'device_id'

Tkd-Alex committed:
set_device and generate_uuids in login method

Tkd-Alex committed:
Merge remote-tracking branch 'upstream/login-flow' into login-flow

Tkd-Alex committed:
Recovery signal atexit

Tkd-Alex committed:
remove again signal and aexit for trevis issue

Commit:
fixing save checkpoint

maxdevblock committed:
device_id and phone_id

  - Fixed `device_id` generated from random UUID (and not from `username`, `password`)
  - Added `reinstall_app_simulation`
  - Added `change_device_simulation`

ohld committed:
Merge branch 'login-flow' into login-flow

Tkd-Alex committed:
Remove save and load cookie.txt file

Tkd-Alex committed:
Move/refactory sync_* methods

Tkd-Alex committed:
Move/refactory devices/uuids methods

Tkd-Alex committed:
Move/refactory pre/login flows methods

Tkd-Alex committed:
Missing imports.. implements all self methods in api.py

Tkd-Alex committed:
Move/refactory some methods

Tkd-Alex committed:
Just fix

Tkd-Alex committed:
E302 expected 2 blank lines, found 1

ohld commented on instabot/api/api.py:
```diff
@@ -614,6 +593,54 @@ def delete_comment(self, media_id, comment_id):
+    def like(self, media_id):
+        data = self.json_data({'media_id': media_id})
+        url = 'media/{media_id}/like/'.format(media_id=media_id)
+        return self.send_request(url, data)
```
it can be:
``` python
return self.send_request(
    url='media/{media_id}/like/'.format(media_id=media_id),
    data=self.json_data({'media_id': media_id}),
)
```

maybe will be more readable

Tkd-Alex replied:
We can do the same thing also for the other methods (for example):
- delete_media
- comment
- . . .

Tkd-Alex committed:
Test fix

ohld commented on instabot/api/api.py:
```diff
@@ -130,81 +98,44 @@ def log_attribution(self, usage='default'):
+    def save_uuid_and_cookie(self):
+        return save_uuid_and_cookie(self)
+
+    def login(self, username=None, password=None, force=False, proxy=None, use_cookie=False, cookie_fname=None, ask_for_code=False, set_device=True, generate_all_uuids=True):
```
too long line

maxdevblock committed:
Too long line fix

maxdevblock committed:
Fix device_id in login

Back to old `device_id` generated from `username` and `password` till we find a fix to change `device_id`

Tkd-Alex committed:
update Save, Like, Comment with radio_type, device_id and other updated field for each req

Tkd-Alex committed:
Merge branch 'login-flow' of github.com:Tkd-Alex/instabot into login-flow

Tkd-Alex committed:
Replace 'id' instead 'pk'

Tkd-Alex committed:
For the moment simulate the like from feed.

Tkd-Alex committed:
Travis Fix1

Tkd-Alex committed:
Continue with fix and replacing pk with id

Tkd-Alex committed:
space travis :)

Tkd-Alex committed:
just fix pk, id

Tkd-Alex committed:
Replace pk with id

Tkd-Alex committed:
Fix the last bulk replace =_=

Tkd-Alex committed:
Shoul be fixed now. Tested locally

Tkd-Alex committed:
Merge pull request #2 from lexan73/pr-1008/Tkd-Alex/login-flow

fixing save checkpoint

maxdevblock committed:
update like_hashtag

Updated like_hashtag to include
hashtag_name, hashtag_id and feed_position

Prepared like and like_medias with kwargs for
like_timeline
like_user
like_location

maxdevblock committed:
Fixes

Fixed syntax and
test_bot_like.py def test_like_hashtag

maxdevblock committed:
Default container_module

Set default container module for like to
container_module="feed_short_url"

Update .gitignore to ignore qiskit junkies

maxdevblock committed:
Default 'is_carousel_bumped_post'

Setting default
'is_carousel_bumped_post': 'false'

TODO: understand when it's 'true'

maxdevblock committed:
Fixed device_id

Commented out
self.device_id = self.uuid

TODO: double check which calls need
'device_id': self.uuid
and which
'device_id': self.device_id

maxdevblock committed:
Fix feed_position in def like

feed_position from int to str

maxdevblock committed:
Like post from link emulation

Added def open_instagram_link in api.py

if like from link emulate a real app:
1. open link
2. get media info
3. like media

maxdevblock committed:
Fix checkpoint_challenge_required

Fix return False if checkpoint_challenge_required and ask_for_code = False

maxdevblock committed:
Fix follow post data

Added missing fields in follow post data

maxdevblock committed:
Fixed unfollow and unlike

Fixed unfollow and unlike
adding missing fields in post data

maxdevblock committed:
Fix like_comment and unlike_comment

Added missing fields in
like_comment and unlike_comment

maxdevblock committed:
New Bot() default parameters

Lighter default filter

ohld commented on instabot/bot/bot_like.py:
```diff
@@ -11,13 +17,30 @@ def like(self, media_id, check_media=True):
+            self.api.open_instagram_link(link)
+            self.logger.debug("Getting media info...")
+            self.api.media_info(media_id)
+        _r = self.api.like(
```
New line -
More readable

ohld commented on instabot/bot/bot_like.py:
```diff
@@ -118,7 +154,18 @@ def like_hashtag(self, hashtag, amount=None):
+    return self.like_medias(
+        medias,
+        container_module="feed_contextual_hashtag",
+        hashtag_name=hashtag, hashtag_id=hashtag_id)
```
Usually the last “)” goes to the next line. And it looks awesome.

maxdevblock committed:
Fix signal if login is threaded

if login function is run threaded, for example in scheduled job, signal will fail because it 'only works in main thread'.
In this case, you may want to call login(is_threaded=True).

basnijholt committed:
add .pre-commit-config.yaml

basnijholt committed:
run pre-commit --run-all

basnijholt committed:
fix all E402 flake8 errors

basnijholt committed:
fix all C901 flake8 errors

basnijholt committed:
sort all imports using isort

basnijholt committed:
add pre-commit to Travis CI

basnijholt committed:
run black

basnijholt committed:
remove pyflakes because we now use flake8

basnijholt committed:
define __all__

basnijholt committed:
remove pycodestyle

maxdevblock committed:
Commented out open link simulation

commented out simulation of "open link in app"

maxdevblock committed:
Fixed login

New defaults
  - use_cookie=True
  - is_threaded=False

maxdevblock committed:
temporary fix test_bot login

temporary fix for test_bo
login(use_cookie=False)

TODO: implement test_bot login with use_cookie=True

maxdevblock committed:
Load uuid and/or cookie

Separate load_uuid from load_cookie

caffeinum reviewed (commented):
👀

maxdevblock committed:
pending threads update

1. fixed post data in approve_pending_thread
2. added endpoint for hide_pending_thread
3. added endpoint for decline_pending_thread

Tkd-Alex committed:
Custom log_filename, choose if you want to save log or not

Tkd-Alex committed:
black pre-commit api file

steffanjensen commented on instabot/api/api.py:
```diff
@@ -46,7 +46,7 @@
 
 class API(object):
-    def __init__(self, device=None, base_path=""):
+    def __init__(self, device=None, base_path="", save_logfile=True, log_filename=None):
```
is it possible to change path by doing
log_filename="path/instabot.log"
or is this only for filename change ?

Tkd-Alex replied:
also work with the path :smile:
